sched/definitions.go

```diff
@@ -24,3 +24,26 @@ type Command struct {
 	// We may extend this with more configuration,
 	// E.g. environment variables
 }
+
+// Status for Job & Tasks
+type Status int
+
+const (
+	// NotRunning, waiting to be scheduled
+	NotStarted Status = iota
+
+	// Currently Scheduled and In Progress Job/Task
+	InProgress
+
+	// Successfully Completed Job/Task
+	Completed
+
+	// Job was Aborted, Compensating Tasks are being Applied.
+	// A RollingBack task has not finished its compensating
+	// tasks yet.
+	RollingBack
```

Comment:
So first off a user cannot Abort a Saga.  Just like a user cannot Abort a Transaction.  User cancelled Jobs is a feature we can add down the line and would require additional status codes to be added.

To answer your question if a Saga is Aborted, then the Job is RollingBack and all the Tasks will be in status RollingBack, Even if their compensating action hasn't started.  Once the compensating task completes then the task is RolledBack state.  When all compensating tasks complete the Job is in the RolledBack state.  I felt this was the clearest way to convey that we are no longer making forward progress on the Job.   Open to suggestions if that's confusing.  

RollingBack indicates that the Saga was aborted for some reason and we are in the process of cleaning up the job.  Since none of these actions are Atomic you do have intermediary states where some tasks are RolledBack and some are still RollingBack.

The property based tests are a great indication of the whole state transition from Saga to Job/Task Status.  

@dbentley I started with just the ForwardRecovery case but then decided it was easy to handle all states.  Also helped me think through it.  Jobs with Rollback Recovery have Ready(Not Started) Running (In Progress), Aborted (RollingBack), Done(Completed/RolledBack) so the user can differentiate between the successful and unsuccessful branch